audit/conn/domainsocket.go

```diff
@@ -0,0 +1,77 @@
+//go:build linux || darwin
+
+package conn
+
+import (
+	"fmt"
+	"net"
+	"os"
+
+	"github.com/microsoft/go-otel-audit/audit/conn/internal/writer"
+)
+
+// Compile check on interface implementation.
+var _ Audit = DomainSocketConn{}
+
+// DomainSocketConn represents a connection to a remote audit server via a unix domain socket
+// set to SOCK_STREAM. This implements conn.Audit interface.
+type DomainSocketConn struct {
+	*writer.Conn
+}
+
+// Type returns the type of the audit connection.
+func (DomainSocketConn) Type() Type {
+	return TypeDomainSocket
+}
+
+func (DomainSocketConn) private() {}
+
+type dsOptions struct {
+	path string
+}
+
+func (o *dsOptions) setDefaults() {
+	const (
+		asaSocket    = "/var/run/mdsd/asa/default_fluent.socket"
```

Comment:
Are these magical values defined by other tools? I'm assuming they _are_, and if so it might be nice to have a comment here explaining where they came from (if we can).

Reply:
They are both mdsd listening sockets for fluent, but the why for both of them ....????  

I'm out getting an answer from my partner team, I'll update this once I get an answer.